src/org/scratchOrgInfoGenerator.ts

```diff
@@ -271,33 +271,21 @@ export const getScratchOrgInfoPayload = async (options: {
   warnings: string[];
 }> => {
   let warnings: string[] = [];
```

Comment:
this is another big section of logic changes that should be checked

the types are a bit of a mess (lowercase vs. capitalized keys)
some of utility methods expect `AnyJson` or such vagueries, and don't like concrete types very much.